azimuth/dataset_split_manager.py

```diff
@@ -153,6 +152,21 @@ def _load_dataset_split(self) -> Optional[Tuple[Dataset, Dataset]]:
             return ds, malformed
         return None
 
+    def _load_base_dataset_split(self, dataset_split) -> Tuple[Dataset, Dataset]:
+        base_dataset_split, malformed_dataset = self._split_malformed(dataset_split)
+
+        # Checking if a persistent id was provided.
+        persistent_id = self.config.columns.persistent_id
+        if persistent_id != DatasetColumn.row_idx:  # Default value
+            if persistent_id not in base_dataset_split.column_names:
+                raise ValueError(f"Persistent ID named {persistent_id} not found in the dataset.")
+            else:
+                all_persistent_ids = base_dataset_split[persistent_id]
+                if len(all_persistent_ids) > len(set(all_persistent_ids)):
+                    raise ValueError(f"Persistent IDs in {persistent_id} column need to be unique.")
```

Comment:
No need for the `else`, if you wish to save an indent.
```suggestion

            all_persistent_ids = base_dataset_split[persistent_id]
            if len(all_persistent_ids) > len(set(all_persistent_ids)):
                raise ValueError(f"Persistent ids in {persistent_id} column need to be unique.")
```